scipy/sparse/dia.py

```diff
@@ -376,6 +376,19 @@ def _with_data(self, data, copy=True):
         else:
             return dia_matrix((data,self.offsets), shape=self.shape)
 
+    def resize(self, shape):
+        if not isshape(shape, nonneg=True):
+            raise TypeError("shape must be a 2-tuple of positive integers")
+        # TODO: perform without conversion to coo
```

Comment:
We could also just leave this one out and let it raise the `NotImplementedError`.

Reply:
What for?

Reply:
I guess what I really should have suggested is raising a `SparseEfficiencyWarning`, because this is likely to be way slower than a user might expect. But there's plenty of precedent for routing through other formats, so I think it's fine to leave things as-is.

Reply:
I think we use SparseEfficiencyWarning for automatic densification. This operation keeps the data sparse. I think it's fine, although I admit there could be a better solution.

Reply:
I think I have an implementation in 7 lines... testing.